### PR TITLE
Apply required changes for showcase template

### PR DESCRIPTION
**Pull request by wtrocki**
@jhellar Tested this changes using different xcode versions (xcodeselect command). It works without much problems! UI looks amazing. Really classy work. 

Problems:
1) repo seems to be overly large (~50mb) that can slow down verification and PR work. This should be used by users who may have slow internet connection.
Can we strip out early commits that have no value now?

2) Swiftlint reports large number of issues. Some of them are critical and should be addressed. Releasing template with such number of issues is not desired:
```
Done linting! Found 3559 violations, 350 serious in 175 files.
```

We can just create linting profile (as we have in SDK and this will reduce number of issues so we can resolve just some of them)

3) Swiftformat - Format we use it's not recommended. It's actually really easy to fix but it will cause all 175 files to be changed.

4) CircleCi needs to be updated to CircleCi2. I have done it but recommended build scripts fails due to issues.

5) No ability to build dev changes - it's not possible to merge any PR now that has dev changes. We need to do the same we have done for JS sdk.

@wei-lee @TommyJ1994 I'm happy to fix all of the issues, but wanted to get your opinion.

## Provisioning

Provisioning profile management in templates is usually set to automatic.
I also changed id to the one that is already configured for RedHat team.
This will ensure that things will work automatically when running in-house.

@wei-lee - This change may be not aligned with strategy you wanted to employ with templates so I'm ok to reject that, but from my opinion this will make things much simpler for both in-house and external developers.

### CODE REVIEW

File: ios-showcase-template.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -524,7 +524,7 @@
 				TargetAttributes = {
 					B3E21B3E1FAC956E00082CF7 = {
 						CreatedOnToolsVersion = 9.1;
-						ProvisioningStyle = Manual;
+						ProvisioningStyle = Automatic;
```
Comment (wtrocki) [inline ios-showcase-template.xcodeproj/project.pbxproj:519]: That's the major change here. Rest of the values do not matter and are automatically generated by xcode.

File: Podfile [hunk 2]
```diff
@@ -4,6 +4,7 @@
 target 'ios-showcase-template' do
   # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
   use_frameworks!
+  platform :ios, '10.0'
```
Comment (wtrocki) [inline Podfile:7]: Default platform is required and crucial to set base for publishing and platform support. This value should go up with every release of the IOS platform.

File: ios-showcase-template.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -918,15 +922,17 @@
...
 				DEVELOPMENT_TEAM = GHPBX39444;
 				INFOPLIST_FILE = "ios-showcase-template/Info.plist";
+				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = "org.aerogear.ios-showcase-template";
+				PRODUCT_BUNDLE_IDENTIFIER = org.aerogear.ShowcaseTemplate;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE = "c708fd4a-a5c7-4353-b315-5d70039e67ce";
-				PROVISIONING_PROFILE_SPECIFIER = RHTINHOUSE;
+				PROVISIONING_PROFILE = "";
```
Comment (wtrocki) [inline ios-showcase-template.xcodeproj/project.pbxproj]: Another important change is to not lock on provisioning profile. Xcode will pick that up automatically depending if dev has created his own developer account or has access to internal one etc.

**Comment by wtrocki** [comment]
Updated

**Comment by wtrocki** [comment]
Resolving also problem with outdated cocoapods lib.

**Comment by wtrocki** [comment]
FIXED 2,3,4. Going to create separate issue for 5. 

@TommyJ1994 @wei-lee Are you happy to strip out repo from large file commits? See point nr1? No strong feelings against that etc.

File: ios-showcase-template.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -897,15 +899,17 @@
...
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_TEAM = GHPBX39444;
 				INFOPLIST_FILE = "ios-showcase-template/Info.plist";
+				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = "org.aerogear.ios-showcase-template";
+				PRODUCT_BUNDLE_IDENTIFIER = org.aerogear.ShowcaseTemplate;
```
Comment (jhellar) [inline ios-showcase-template.xcodeproj/project.pbxproj]: We should probably change also this [1] and maybe [2] (not sure if it is used anywhere).

[1] https://github.com/aerogear/ios-showcase-template/blob/83f5a26c615f1dc4b27abf5dc978a9434b02c842/ios-showcase-template/di/AppComponents.swift#L35
[2] https://github.com/aerogear/ios-showcase-template/blob/4f3ac372d06ca512d54fb1be86bc412ee29e6c37/ios-showcase-template/Info.plist#L26

File: .swiftlint.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+disabled_rules: # rule identifiers to exclude from running
+included: # paths to include during linting. `--path` is ignored if present.
+  - ios-showcase-template
+excluded: # paths to ignore during linting. Takes precedence over `included`.
+  - /example/Pods
```
Comment (jhellar) [inline .swiftlint.yml]: Can this line be changed to `- /Pods` or removed?

**Review by jhellar — COMMENTED** [review]
LGTM.

**Review by jhellar — APPROVED** [review]

**Pull request merged**